doc/langref/test_pointer_arithmetic.zig

```diff
@@ -11,6 +11,9 @@ test "pointer arithmetic with many-item pointer" {
     // slicing a many-item pointer without an end is equivalent to
     // pointer arithmetic: `ptr[start..] == ptr + start`
     try expect(ptr[1..] == ptr + 1);
+
+    // subtraction between the memory addresses of any two pointers except slices is supported
+    try expect(&ptr[1] - &ptr[0] == @sizeOf(i32));
```

Comment:
I don't think I understand completely but if you want `&ptr[1] - ptr == 1` behavior how about you divide the result by the byte size of the element type?

Reply:
> You can still get index behavior by dividing.

That's true. Both behaviors make some sense, and dividing / multiplying can help adapting one to the other.

Today (as of 0.13.0) `ptr + x` already consider x as number of array elements. It could seem pretty surprising and sort of confusing if `ptr + x - ptr != x` .

> I guess I can't really answer this but maybe a maintainer can once some other reviews this.

Good point.

Reply:
Oh, I guess I can see why that might be confusing. I knew that before implementing this but didn't think much of it because arithmetic operators on pointers are special and you won't ever use them as often as arithmetic on integers so you might check the documentation before using them.
Ultimately I just implemented a proposal that is accepted so I blindly expected it to be well thought out.

Reply:
So you also have to make sure both the LHS and RHS pointers' element types are equal right?